File: L005/Next-Patterns/pattern20.java
import java.util.Scanner;

public class pattern20 {

    public static Scanner sc = new Scanner(System.in);

    public static void pattern20_(int n) {

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {

                if (j == 0 || j == n - 1) {
                    System.out.print("*\t");
                }else if(i>=n/2 && ( i == j || i+j == n-1)){
                    System.out.print("*\t");
                }   
                else {
                    System.out.print("\t");
                }
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {

        int n = sc.nextInt();
        pattern20_(n);
    }
}

File: L005/Next-Patterns/pattern20.cpp
#include<iostream>
using namespace std;

void pattern20_(int n) {

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {

                if (j == 0 || j == n - 1) {
                    cout<<"*\t";
                }else if(i>=n/2 && ( i == j || i+j == n-1)){
                    cout<<"*\t";
                }   
                else {
                    cout<<"\t";
                }
            }
            cout<<endl;
        }

}       
    int main(){

        int n;
        cin>>n;

        pattern20_(n);

        return 0;
    }
